spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/sql/TestDropTable.java

```diff
@@ -0,0 +1,260 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.sql;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.spark.SparkCatalog;
+import org.apache.iceberg.spark.SparkCatalogTestBase;
+import org.apache.iceberg.spark.SparkSessionCatalog;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestDropTable extends SparkCatalogTestBase {
+  private Map<String, String> config = null;
+  private String implementation = null;
+  private SparkSession session = null;
+
+  @Parameterized.Parameters(name = "catalogName = {0}, implementation = {1}, config = {2}")
+  public static Object[][] parameters() {
+    return new Object[][]{
+        {"testhive", CustomSparkCatalog.class.getName(),
+            ImmutableMap.of(
+                "type", "hive",
+                "default-namespace", "default"
+            )},
+        {"testhadoop", CustomSparkCatalog.class.getName(),
+            ImmutableMap.of(
+                "type", "hadoop"
+            )},
+        {"spark_catalog", CustomSparkSessionCatalog.class.getName(),
+            ImmutableMap.of(
+                "type", "hive",
+                "default-namespace", "default",
+                "parquet-enabled", "true",
+                "cache-enabled", "false" // Spark will delete tables using v1, leaving the cache out of sync
+            )}
+    };
+  }
+
+  public TestDropTable(String catalogName, String implementation, Map<String, String> config) {
+    super(catalogName, implementation, config);
+    this.config = config;
+    this.implementation = implementation;
+  }
+
+  @Before
+  public void createTable() {
+    // Spark CatalogManager cached the loaded catalog, here we use new SparkSession to force it load the catalog again
```

Comment:
@rdblue please see the comments. We have customed the `catalog` to verify whether `drop` or `purge` method is called. So we need to load a new catalog for each test. That's why we define a new `drop` method.

Reply:
This doesn't sound correct to me. You should not need to alter the catalog to check whether a table is purged. Can't you get the location from the table itself? And this appears to make the tests specific to the Hadoop catalog.

Reply:
This adopted the previous comments, maybe it is too old:

> This isn't sufficient to test purge. I think you should create a mock catalog to test that the argument is passed through to dropTable correctly.